modules/education/src/main/java/edu/stanford/spezi/modules/education/videos/EducationScreen.kt

```diff
@@ -0,0 +1,285 @@
+                            title = videoSection.title,
+                            description = videoSection.description,
+                            videos = videoSection.videos,
+                            expandedStartValue = videoSection.isExpanded,
+                            onExpand = {
+                                videoSection.isExpanded = !videoSection.isExpanded
+                            },
+                            onActionClick = { video ->
+                                onAction(
+                                    Action.VideoSectionClicked(
+                                        video = video
+                                    )
+                                )
+                            }
+                        )
+                    }
+                }
+            }
+        }
+    }
+}
+
+private class EducationScreenPreviewProvider :
+    PreviewParameterProvider<Pair<EducationUiState, (Action) -> Unit>> {
+    val factory = EducationUiStateFactory()
+    private val errorState = factory.createErrorState("An error occurred")
+    private val loadingState = factory.createLoadingState()
+    private val successState =
+        factory.createSuccessState(listOf(getVideoSection(), getVideoSection()))
+    override val values: Sequence<Pair<EducationUiState, (Action) -> Unit>> = sequenceOf(
+        Pair(errorState) { },
+        Pair(loadingState) { },
+        Pair(successState) { }
+    )
+
+    override val count: Int = values.count()
+}
+
+@Preview(showBackground = true)
+@Composable
+private fun EducationScreenPreview(
+    @PreviewParameter(EducationScreenPreviewProvider::class) params: Pair<EducationUiState, (Action) -> Unit>,
+) {
+    SpeziTheme {
+        EducationScreen(
+            uiState = params.first,
+            onAction = params.second
+        )
+    }
+}
+
+private class EducationUiStateFactory {
+    fun createErrorState(errorMessage: String): EducationUiState {
+        return EducationUiState(
+            loading = false,
+            error = errorMessage
+        )
+    }
+
+    fun createLoadingState(): EducationUiState {
+        return EducationUiState(
+            loading = true,
+            error = null
+        )
+    }
+
+    fun createSuccessState(videoSections: List<VideoSection>): EducationUiState {
+        return EducationUiState(
+            loading = false,
+            error = null,
+            videoSections = videoSections
+        )
+    }
+}
+
+private fun getVideoSection(): VideoSection {
+    return VideoSection(
+        title = "Medications Videos",
+        description = "Videos about medications",
+        videos = listOf(
+            Video("Video Title 1", "Video Description 1", youtubeId = "W3R_ETKMj0E"),
+            Video("Video Title 2", "Video Description 2", youtubeId = "W3R_ETKMj0E"),
+            Video("Video Title 3", "Video Description 3", youtubeId = "W3R_ETKMj0E")
+        )
+    )
+}
```

Comment:
If you recall a proposal of mine in the very beginning to move the previews and preview parameters in a separate dedicated package -> this was the reason, so that we could just exclude those file from code coverage. Maybe we can rethink this in the future (see this [issue](https://github.com/jacoco/jacoco/issues/1208) on jacocco)

Reply:
if there is no other solution, we can do it like this 👍